contracts/src/entry/library.cairo

```diff
@@ -75,6 +102,37 @@ namespace Entries {
         return (mean_price,);
     }
 
+    func generic_entries_median{range_check_ptr}(entries_len: felt, entries: GenericEntry*) -> (
+        value: felt
+    ) {
+        let (sorted_entries) = sort_generic_entries_by_value(entries_len, entries);
+
+        let (q, r) = unsigned_div_rem(entries_len, 2);
+        let is_even = 1 - r;
+
+        if (is_even == FALSE) {
+            let median_idx = entries_len - q - 1;  // 0-indexed
+            let median_entry = [sorted_entries + median_idx * GenericEntry.SIZE];
+            return (median_entry.value,);
+        }
+
+        let median_idx_1 = entries_len - q - 1;
+        let median_entry_1 = [sorted_entries + median_idx_1 * SpotEntry.SIZE];
```

Comment:
GenericEntry.SIZE

Reply:
Think we can actually use `sorted_entries[median_idx_1]` and then don't have to worry about this